service/matching/tasklist/task_reader.go

```diff
@@ -212,6 +215,14 @@ getTasksPumpLoop:
 
 				if len(tasks) == 0 {
 					tr.taskAckManager.SetReadLevel(readLevel)
+
+					if tr.taskAckManager.GetAckLevel() == initialReadLevel {
+						// Even though we didn't handle any tasks, we want to advance the ack-level
+						// in order to avoid needless querying database the next time.
+						// This is safe since we started reading exactly from the current AckLevel and read no tasks
+						tr.taskAckManager.SetAckLevel(readLevel)
```

Comment:
Ack level is persisted every 1m (`MatchingUpdateAckInterval`) and it seems too late. I'd consider setting it to < 10s since we now know moving ack level to the accurate position matters.

Reply:
I didn't know we also save it periodically. Only saw metric emission on updateAckTimer. Good finding!
I think it doesn't matter so much for the fix since we `persistAckLevel` at Stop() of `TaskListReader` anyway.